src/routers/router.php

```diff
@@ -0,0 +1,186 @@
+<?php
+
+namespace Nectary;
+
+use \Nectary\Factories\Dependency_Injection_Factory;
+
+/**
+ * The base presenter has functionality to route
+ * undefined methods to other class methods.
+ *
+ * In order to make use of this functionality, simply
+ * define a `protected $route` in your class that
+ * implements this abstract class.
+ *
+ * Routes are expected to have:
+ * - to : The Class and Method to route to.
+ * - expects : The arguments expected by this route.
+ *             Order is important, incoming arguments
+ *             are named in this order.
+ * - on_error : If a Request has a validation error
+ *              it will call this function with a $message,
+ *              and $this.
+ *
+ * Example:
+ *
+ * ```
+ * $present = new Events_Presenter();
+ * $present->get_blurb_list( [ 'slug' ], 2005 );
+ * ```
+ *
+ * Will get mapped to:
+ *
+ * ```
+ * array(
+ *   'category_slugs' => [ 'slugs' ],
+ *   'year' => 2005
+ * )
+ * ```
+ *
+ * When given the following route:
+ *
+ * ```
+ * protected $routes = [
+ *   'get_blurb_list' => [
+ *    'to'       => '\Gios_Api\Events_Controller@get_blurb_list',
+ *     'expects'  => [ 'category_slugs', 'year', 'base_path' ],
+ *     'on_error' => [ 'Events_Presenter', 'error_callback' ],
+ *   ],
+ * ];
+ * ```
+ *
+ * Purposefully not in the Gios_Api namespace.
+ */
+abstract class Router {
```

Comment:
@rraub I'm calling this a Router instead of a Presenter. Our current presenters in Gios API can extend Router instead of extending Presenter, ex: `class Event_Presenter extends Router`.